jdaviz/core/loaders/importers/image/image.py

```diff
@@ -0,0 +1,230 @@
+import os
+import asdf
+
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from astropy.nddata import NDData
+from astropy.wcs import WCS
+from glue.core.data import Component, Data
+from traitlets import Bool, List, Unicode, observe, Any
+
+
+from jdaviz.core.template_mixin import AutoTextField, SelectFileExtensionComponent
+
+from jdaviz.core.registries import loader_importer_registry
+from jdaviz.core.loaders.importers import BaseImporterToDataCollection
+
+from jdaviz.utils import (
+    standardize_metadata, standardize_roman_metadata,
+    PRIHDR_KEY, _wcs_only_label, download_uri_to_path, get_cloud_fits
+)
+
+try:
+    from roman_datamodels import datamodels as rdd
+except ImportError:
+    HAS_ROMAN_DATAMODELS = False
+else:
+    HAS_ROMAN_DATAMODELS = True
+
+
+__all__ = ['ImageImporter']
+
+
+@loader_importer_registry('Image')
+class ImageImporter(BaseImporterToDataCollection):
+    template_file = __file__, "./image.vue"
+
+    # HDUList-specific options
+    input_hdulist = Bool(False).tag(sync=True)
+    extension_items = List().tag(sync=True)
+    extension_selected = Any().tag(sync=True)
+    extension_multiselect = Bool(True).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.app.config == 'imviz':
+            self.data_label_default = 'Image'
+
+        self.input_hdulist = isinstance(self.input, fits.HDUList)
+        if self.input_hdulist:
+            self.extension = SelectFileExtensionComponent(self,
+                                                          items='extension_items',
+                                                          selected='extension_selected',
+                                                          multiselect='extension_multiselect',
+                                                          manual_options=self.input,
+                                                          filters=[_validate_fits_image2d])
+            self.extension.selected = [self.extension.choices[0]]
```

Comment:
this might not be necessary, or might be able to call `self.extension.select_default()` instead